src/demo/player.rs

```diff
@@ -10,62 +6,57 @@ use crate::{
         animation::PlayerAnimation,
         movement::{MovementController, ScreenWrap},
     },
-    screens::Screen,
     AppSet,
 };
 
 pub(super) fn plugin(app: &mut App) {
-    app.observe(spawn_player);
     app.register_type::<Player>();
 
-    // Record directional input as movement controls.
     app.add_systems(
         Update,
         record_player_directional_input.in_set(AppSet::RecordInput),
     );
 }
 
-#[derive(Event, Debug)]
-pub struct SpawnPlayer;
-
-#[derive(Component, Debug, Clone, Copy, PartialEq, Eq, Default, Reflect)]
+/// A marker component for player entities.
+#[derive(Component, Reflect, Clone, Copy, PartialEq, Eq, Debug)]
 #[reflect(Component)]
 pub struct Player;
 
-fn spawn_player(
-    _trigger: Trigger<SpawnPlayer>,
-    mut commands: Commands,
-    image_handles: Res<ImageHandles>,
-    mut texture_atlas_layouts: ResMut<Assets<TextureAtlasLayout>>,
-) {
-    // A texture atlas is a way to split one image with a grid into multiple
-    // sprites. By attaching it to a [`SpriteBundle`] and providing an index, we
-    // can specify which section of the image we want to see. We will use this
-    // to animate our player character. You can learn more about texture atlases in
-    // this example: https://github.com/bevyengine/bevy/blob/latest/examples/2d/texture_atlas.rs
+/// Construct a player entity.
```

Comment:
```suggestion
/// Construct a player entity by inserting the relevant components.
```

Reply:
This could add children / do other things as well.

Reply:
I would like to guide users more on what "construct" means in this context and why there is a `Entity` as a parameter.